app/controllers/concerns/defence_request_authorization.rb

```diff
@@ -1,19 +1,24 @@
 module DefenceRequestAuthorization
 
+  private
+
   def find_defence_request
+    defence_request_id = params[:defence_request_id] || params[:id]
     @defence_request = DefenceRequest.find(defence_request_id)
   end
 
-  def set_policy_with_context
-    @policy ||= policy(PolicyContext.new(defence_request, current_user))
-  end
-
   def defence_request
     @defence_request ||= DefenceRequest.new
   end
 
   def new_defence_request_form
```

Comment:
In that case I'd personally prefer to have it in the `BaseController` rather than here. Because this is an authorisation concern only. 
But I'll leave this up to you, maybe ask another opinion. I'm not insisting on this to be done now :)